src/gpio_monitor.cpp

```diff
@@ -23,7 +27,26 @@ void GpioEventHandler::setEventHandlerForGpioPresence(
 void GpioMonitor::initHandlerForGpio(
     const std::shared_ptr<boost::asio::io_context>& i_ioContext)
 {
-    // ToDo Add implementation.
-    (void)i_ioContext;
+    try
+    {
+        std::vector<types::GpioPollingParameters>
+            l_pollingRequiredFrusParamsList =
+                jsonUtility::getListOfPollingParamsForFrus(m_sysCfgJsonObj);
+
+        for (const auto& l_pollingParameters : l_pollingRequiredFrusParamsList)
```

Comment:
what would happen if an empty tuple is the output from jsonUtility::getListOfPollingParamsForFrus(m_sysCfgJsonObj); ?

Reply:
since it is an empty list, will not enter into the loop

Reply:
ok got it.